pkg/client/client_venconn_test.go

```diff
@@ -0,0 +1,381 @@
+		}))
+		require.NoError(t, kclient.Create(context.Background(), &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{Name: "venafi-connection", Namespace: "venafi"},
+		}))
+		require.NoError(t, kclient.Create(context.Background(), &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{Name: "venafi-connection-role"},
+			Rules: []rbacv1.PolicyRule{
+				{APIGroups: []string{""}, Resources: []string{"namespaces"}, Verbs: []string{"get", "list", "watch"}},
+				{APIGroups: []string{"jetstack.io"}, Resources: []string{"venaficonnections"}, Verbs: []string{"get", "list", "watch"}},
+				{APIGroups: []string{"jetstack.io"}, Resources: []string{"venaficonnections/status"}, Verbs: []string{"get", "patch"}},
+			},
+		}))
+		require.NoError(t, kclient.Create(context.Background(), &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{Name: "venafi-connection-rolebinding"},
+			RoleRef:    rbacv1.RoleRef{APIGroup: "rbac.authorization.k8s.io", Kind: "ClusterRole", Name: "venafi-connection-role"},
+			Subjects:   []rbacv1.Subject{{Kind: "ServiceAccount", Name: "venafi-connection", Namespace: "venafi"}},
+		}))
+		require.NoError(t, kclient.Create(context.Background(), &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{Name: "accesstoken", Namespace: "venafi"},
+			StringData: map[string]string{"accesstoken": "VALID_ACCESS_TOKEN"},
+		}))
+		require.NoError(t, kclient.Create(context.Background(), &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{Name: "apikey", Namespace: "venafi"},
+			StringData: map[string]string{"apikey": "VALID_API_KEY"},
+		}))
+		require.NoError(t, kclient.Create(context.Background(), &rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{Name: "venafi-connection-secret-reader", Namespace: "venafi"},
+			Rules: []rbacv1.PolicyRule{
+				{APIGroups: []string{""}, Resources: []string{"secrets"}, Verbs: []string{"get"}, ResourceNames: []string{"accesstoken", "apikey"}},
+			},
+		}))
+		require.NoError(t, kclient.Create(context.Background(), &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{Name: "venafi-connection-secret-reader", Namespace: "venafi"},
+			RoleRef:    rbacv1.RoleRef{APIGroup: "rbac.authorization.k8s.io", Kind: "Role", Name: "venafi-connection-secret-reader"},
+			Subjects:   []rbacv1.Subject{{Kind: "ServiceAccount", Name: "venafi-connection", Namespace: "venafi"}},
+		}))
+
+		test.given = strings.ReplaceAll(test.given, "FAKE_VENAFI_CLOUD_URL", fakeVenafiCloud.URL)
+		test.given = strings.ReplaceAll(test.given, "FAKE_TPP_URL", fakeTPP.URL)
+		for _, obj := range parse(test.given) {
+			require.NoError(t, kclient.Create(context.Background(), obj))
+		}
+
+		err = cl.PostDataReadingsWithOptions([]*api.DataReading{}, client.Options{ClusterName: "test cluster name"})
+		if test.expectErr != "" {
+			assert.EqualError(t, err, test.expectErr)
+		}
+
+		got := v1alpha1.VenafiConnection{}
+		kclient.Get(context.Background(), types.NamespacedName{Name: "venafi-components", Namespace: "venafi"}, &got)
```

Comment:
Unchecked error:

```console
$ golangci-lint run ./pkg/client/...
...
pkg/client/client_venconn_test.go:198:14: Error return value of `kclient.Get` is not checked (errcheck)
                kclient.Get(context.Background(), types.NamespacedName{Name: "venafi-components", Namespace: "venafi"}, &got)
                           ^
```

Reply:
Thanks for pointing that out, I had started running golangci-lint as I was working in #556 but didn't start fixing the warnings. I will fix this.